components/script/dom/element.rs

```diff
@@ -2402,7 +2412,7 @@ impl<'a> ::selectors::Element for Root<Element> {
 
             // FIXME(#15746): This is wrong, we need to instead use extended filtering as per RFC4647
             //                https://tools.ietf.org/html/rfc4647#section-3.3.2
-            NonTSPseudoClass::Lang(ref lang) => lang.eq_ignore_ascii_case(&self.get_lang()),
+            NonTSPseudoClass::Lang(ref lang) => self.is_lang_matches(lang),
```

Comment:
If you simply call `lang_check(&*self.get_lang(), &*lang)`, you can eliminate the `is_lang_matches` functions.